src/Application.vala

```diff
@@ -27,11 +27,13 @@ namespace Maya {
 
     public class Application : Gtk.Application {
         public MainWindow window;
-        public static GLib.Settings saved_state;
+        public static GLib.Settings? saved_state = null;
```

Comment:
What happens when this is null? There are a bunch of cases where it's used without any null checks. See lines 73, 117, 128, 150.

Reply:
I admit I was only envisaging this being `null` during (current) testing.  If you want to cater for full operation without any settings the whole app will have to be reviewed and modified.

Reply:
Makes sense. I'm worried that if we run into something in our tests that requires this file, then it will probably break. If we want to test the main application code, instead of just `core`, we'll have to deal with this in some way. That may be a question for a different time though.

It could also lead to weird issues if someone somehow gets Calendar installed and the settings install fails. I don't know if that could really happen (outside of our CI server), so I trust you if you say this is safe.

Reply:
I am not saying those are not valid concerns but they exist already in master so they can be left for another PR rather than expand the scope of this one. I just wanted to get this PR ready to merge.